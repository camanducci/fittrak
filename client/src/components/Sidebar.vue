<template>
  <v-navigation-drawer app persistent clipped v-model="drawer">
  <ul>
    <li>
      <router-link to="/home">
        Track 
      </router-link>
    </li>

    <li>
      <router-link to="/progress">
        Progress
      </router-link>
    </li>

    <li>
      <router-link to="/history">
        History
      </router-link>
    </li>

    <li>
      <router-link to="/settings">
        Settings
      </router-link>
    </li>

    <li>
      <a href="" @click.prevent="logout">
        <span class="icon is-small"><i class="fa fa-sign-out" aria-hidden="true"></i></span>
        Logout
      </a>
    </li>
  </ul>


</v-navigation-drawer>
</template>

<script>
import auth from '../auth'

export default {
  props: ['drawer'],
  methods: {
    logout () {
      return auth.logout()
    }
  }
}
</script>
