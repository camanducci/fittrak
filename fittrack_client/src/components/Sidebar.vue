<template>
<div class="sidebar" v-bind:class="showSidebar">
  <div class="toggle-menu" v-on:click="toggleSidebar">
    <button>
      <span class="icon is-medium"><i class="fa fa-bars"></i></span>
    </button>
  </div>

  <aside class="menu">
    <p class="menu-label">Menu</p>
    <ul class="menu-list">
      <li>
        <a href="">
          <span class="icon is-small"><i class="fa fa-heart" aria-hidden="true"></i></span>
          Track 
        </a>
      </li>

      <li>
        <a href="">
          <span class="icon is-small"><i class="fa fa-pie-chart" aria-hidden="true"></i></span>
          Progress
        </a>
      </li>

      <li>
        <a href="">
          <span class="icon is-small"><i class="fa fa-history" aria-hidden="true"></i></span>
          History
        </a>
      </li>

      <li>
        <a href="">
          <span class="icon is-small"><i class="fa fa-sliders" aria-hidden="true"></i></span>
          Settings
        </a>
      </li>

      <li>
        <a href="">
          <span class="icon is-small"><i class="fa fa-sign-out" aria-hidden="true"></i></span>
          Logout
        </a>
      </li>

    </ul>
  </aside>
</div>

</template>

<style lang="scss" scoped>
$darkgrey: #353535;
$darkgrey2: #444444;
$white: #fffff;

div.sidebar {
  z-index: 2;
  left: 0;
  top: 0;
  height: 100%;
  background: $darkgrey;
  position: fixed;
  color: $white;
  display: none;
}

div.sidebar.active {
  width: 20%;
  display: block;
}

div.sidebar aside.menu {
  padding: 1rem;
}

div.toggle-menu {
  border-bottom: 1px solid $darkgrey2;
}

div.toggle-menu button {
  background: $darkgrey;
  color: #fff;
  border: none;
}

div.toggle-menu button span {
  color: $white;
}
</style>

<script>
export default {
  methods: {
    toggleSidebar () {
      this.$store.commit('toggleSidebar')
    }
  },
  computed: {
    showSidebar () {
      return {
        active: this.$store.state.showSidebar
      }
    }
  }
}
</script>


