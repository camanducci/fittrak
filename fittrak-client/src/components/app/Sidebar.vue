<template>
  <v-list dark>
    <router-link 
      tag="v-list-tile" 
      to="/">
      <v-list-tile-action>
        <v-icon>home</v-icon>
      </v-list-tile-action>
      <v-list-tile-content>
        <v-list-tile-title>Home</v-list-tile-title>
      </v-list-tile-content>
    </router-link>

    <router-link 
      tag="v-list-tile" 
      to="/progress">
      <v-list-tile-action>
        <v-icon>trending_up</v-icon>
      </v-list-tile-action>
      <v-list-tile-content>
        <v-list-tile-title>Progress</v-list-tile-title>
      </v-list-tile-content>
    </router-link>

    <router-link 
      tag="v-list-tile" 
      to="/history">
      <v-list-tile-action>
        <v-icon>history</v-icon>
      </v-list-tile-action>
      <v-list-tile-content>
        <v-list-tile-title>History</v-list-tile-title>
      </v-list-tile-content>
    </router-link>

    <router-link 
      tag="v-list-tile" 
      to="/settings">
      <v-list-tile-action>
        <v-icon>settings</v-icon>
      </v-list-tile-action>
      <v-list-tile-content>
        <v-list-tile-title>Settings</v-list-tile-title>
      </v-list-tile-content>
    </router-link>

    <v-divider />

    <v-list-tile @click="toReleaseNotes">
      <v-list-tile-action>
        <v-icon>notes</v-icon>
      </v-list-tile-action>
      <v-list-tile-content>
        <v-list-tile-title>Release Notes</v-list-tile-title>
      </v-list-tile-content>
    </v-list-tile>

    <v-list-tile @click="logout">
      <v-list-tile-action>
        <v-icon>exit_to_app</v-icon>
      </v-list-tile-action>
      <v-list-tile-content>
        <v-list-tile-title>Sign out</v-list-tile-title>
      </v-list-tile-content>
    </v-list-tile>
  </v-list>
</template>

<script>
export default {
  name: "SidebarNavigationItems",
  methods: {
    toReleaseNotes() {
      window.location.href = "https://github.com/tsoporan/fittrak/releases";
    },
    logout() {
      // Allow server side routing to kick in and logout session
      window.location.replace("/accounts/logout/");
    }
  }
};
</script>
