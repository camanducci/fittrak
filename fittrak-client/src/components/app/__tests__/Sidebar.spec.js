import { shallowMount } from "@vue/test-utils";

import Sidebar from "@/components/app/Sidebar";

describe("Sidebar", () => {
  test("Sidebar renders", () => {
    const component = shallowMount(Sidebar, {
      propsData: {
        viewer: {
          username: "Alice"
        }
      }
    });
    expect(component.html()).toMatchSnapshot();
  });
});
