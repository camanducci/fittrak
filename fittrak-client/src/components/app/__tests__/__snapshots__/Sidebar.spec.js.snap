// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Sidebar renders 1`] = `
<v-list-stub dark="">
  <v-list-tile-stub activeclass="primary--text">
    <v-list-tile-content-stub>
      <v-list-tile-title-stub class="title">
        Heya, <span class="primary--text text--lighten-1">Alice</span></v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-divider-stub></v-divider-stub>
  <router-link-stub to="/" tag="v-list-tile" event="click">
    <v-list-tile-action-stub>
      <v-icon-stub>home</v-icon-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Home</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </router-link-stub>
  <router-link-stub to="/progress" tag="v-list-tile" event="click">
    <v-list-tile-action-stub>
      <v-icon-stub>trending_up</v-icon-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Progress</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </router-link-stub>
  <router-link-stub to="/history" tag="v-list-tile" event="click">
    <v-list-tile-action-stub>
      <v-icon-stub>history</v-icon-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>History</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </router-link-stub>
  <router-link-stub to="/settings" tag="v-list-tile" event="click">
    <v-list-tile-action-stub>
      <v-icon-stub>settings</v-icon-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Settings</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </router-link-stub>
  <v-divider-stub></v-divider-stub>
  <v-list-tile-stub activeclass="primary--text">
    <v-list-tile-action-stub>
      <v-icon-stub>notes</v-icon-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Release Notes</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
  <v-list-tile-stub activeclass="primary--text">
    <v-list-tile-action-stub>
      <v-icon-stub>exit_to_app</v-icon-stub>
    </v-list-tile-action-stub>
    <v-list-tile-content-stub>
      <v-list-tile-title-stub>Sign out</v-list-tile-title-stub>
    </v-list-tile-content-stub>
  </v-list-tile-stub>
</v-list-stub>
`;
